src/tunnel.cc

```diff
@@ -319,7 +316,6 @@ void
 TunnelStateData::serverClosed()
 {
     server.noteClosure();
-    retryOrBail("server closed");
 }
```

Comment:
@eduard-bagdasaryan, it looks like the retryOrBail() call in #489 branch was removed in branch commit be346b3 (when resolving conflicts with master). When you review #957 that effectively restores this call, please see whether you can recall why the call was removed -- if that removal was not an accident, then we may need to do more in that PR!

Reply:
I believe that this removal was a mistake. I could not recollect exactly why I removed it, but perhaps I got confused by TunnelStateData::retriable, which is reset before the TunnelStateData::serverClosed() handler is assigned and makes impossible the 'retry' path, but still requires the 'bail' functionality.